compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -512,6 +512,11 @@ impl<'a> Parser<'a> {
         self
     }
 
+    #[inline]
+    pub fn set_recovery(&mut self, recovery: Recovery) {
```

Comment:
If we're gonna expose a `set_*`, why not just make this a `with_*`, so it's responsible with the resetting? Seems far easier to prevent misuse.

Reply:
Otherwise, there doesn't really seem to be much value of having a setter over just making the `Recovery` field public so it can be mutated directly.